.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -31,6 +31,42 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  get-test-directories:
+    if: "!contains(fromJSON(inputs.skip).jobs, 'terraform-tests')"
+    name: Get Test Directories For OPAL Validation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build matrix
+        id: matrix
+        shell: bash
+        run: |
+          DIRS="$(ls -d .tests/*/)"
+          [[ ${#DIRS} > 0 ]] && echo "::set-output name=directories::[\"${DIRS//$'\n'/\",\"}\"]" || echo "::set-output name=directories::[]"
+    outputs:
+      directories: ${{ steps.matrix.outputs.directories }}
+
+  terraform-test-apply:
+    needs: get-test-directories
+    if: "!contains(fromJSON(inputs.skip).jobs, 'terraform-tests')"
```

Comment:
Isn't this condition implicit in `needs`? There's probably some way to avoid expressing the condition twice.

Reply:
oh clever! 

unfortunately from some testing just now, i've learned that if a build matrix is empty, it doesn't just skip the downstream tests, it causes a failure (which will happen frequently until every repo gets its tests written). So while you're right, i can do without the ` if: "!contains(fromJSON(inputs.skip).jobs, 'terraform-tests')"` here, i had to add a new conditional to make sure the build matrix is not empty. 

(Which, fwiw, that conditional looks like this atm: `if: ${{ needs.get-test-directories.outputs.directories != '[]' && needs.get-test-directories.outputs.directories != '' }}`

checks for `[]` if there are no test directories, and for `''` if the `get-test-directories` job is skipped by the calling workflow. )